blob/src/main/java/io/crate/protocols/http/HttpBlobHandler.java

```diff
@@ -367,22 +380,35 @@ private void fullContentResponse(HttpRequest request, String index, final String
     private ChannelFuture transferFile(final String digest, RandomAccessFile raf, long position, long count)
         throws IOException {
 
-        final FileRegion region = new DefaultFileRegion(raf.getChannel(), position, count);
         Channel channel = ctx.channel();
-        ChannelFuture writeFuture = channel.write(region, ctx.newProgressivePromise());
-        writeFuture.addListener(new ChannelProgressiveFutureListener() {
+        final ChannelFuture fileFuture;
+        final ChannelFuture endMarkerFuture;
+        if (sslEnabled) {
+            HttpChunkedInput httpChunkedInput =
+                new HttpChunkedInput(new ChunkedFile(raf, 0, count, 8192));
```

Comment:
I think it is hard to tell what the optimal size is because it depends on the network environment. I think having larger chunks generally is beneficial for the download speed. We can revisit when we do more benchmarks. If necessary, we can make the default configurable.